docs/02.API-REFERENCE.md

```diff
@@ -3868,6 +3868,121 @@ example (void)
+  is no longer needed.
+- The es.next profile enables this by default.
+
+**Prototype**
+
+```c
+jerry_value_t
+jerry_from_property_descriptor (const jerry_property_descriptor_t *src_prop_desc_p)
+```
+
+- `src_prop_desc_p` - the input property descriptor.
+- return
+  - [jerry_value_t](#jerry_value_t)
+  - jerry_value_t - if success
+  - value marked with error flag - otherwise
+
+**Example**
+
+[doctest]: # (test="compile")
+
+```c
+#include <jerryscript.h>
+
+static void
+example (void)
+{
+  // create a object.
+  jerry_value_t object = jerry_create_object ();
+  // define a string
+  jerry_value_t prop_name = jerry_create_string_from_utf8 ((jerry_char_t *) "value");
+  // create a value to the 'value' property
+  jerry_value_t value = jerry_create_boolean (true);
+
+  // setting 'value' property with the value true on the object variable
+  jerry_set_property (object, prop_name, prop_name);
+
+  // define a property_descriptor
+  jerry_property_descriptor_t prop_desc;
+
+  // getting 'value' property on object variable a store the value in prop_desc
+  jerry_get_own_property_descriptor (object, prop_name, &prop_desc);
+
+  // creating a new object and give a value based on prop_desc
+  jerry_value_t from_object = jerry_from_property_descriptor (&prop_desc);
+
+
+  jerry_release_value (object);
+  jerry_release_value (prop_name);
+  jerry_release_value (value);
+  jerry_release_value (from_object);
+  jerry_free_property_descriptor_fields (&prop_desc);
+}
+```
+
+## jerry_to_property_descriptor
+
+**Summary**
+
+*Notes*:
+- Returned value must be freed with [jerry_release_value](#jerry_release_value) when it
+  is no longer needed.
+- The es.next profile enables this by default.
+
+**Prototype**
+
+```c
+jerry_value_t
+jerry_value_t jerry_to_property_descriptor (jerry_value_t obj_value, jerry_property_descriptor_t *out_prop_desc_p);
+```
+
+- `obj_value` - the input object
+- `src_prop_desc_p` - the output property descriptor.
+- return
+  - [jerry_value_t](#jerry_value_t)
+  - jerry_value_t - if success
+  - value marked with error flag - otherwise
+
+**Example**
+
+[doctest]: # (test="compile")
+
+```c
+#include <jerryscript.h>
+
+static void
+example (void)
+{
+  // create a object.
+  jerry_value_t object = jerry_create_object ();
+  // defined a string
+  jerry_value_t prop_name = jerry_create_string_from_utf8 ((jerry_char_t *) "value");
+  // create a value to the length property
+  jerry_value_t value = jerry_create_boolean (true);
+  // create a property descriptor
+  jerry_property_descriptor_t prop_desc;
+
+  // setting length property with the value true on the object variable
+  jerry_set_property (object, prop_name, prop_name);
+
+  // give values to prop_desc based on object's value
+  jerry_to_property_descriptor (object, &prop_desc);
+
+  jerry_release_value (object);
+  jerry_release_value (prop_name);
+  jerry_release_value (value);
+  jerry_free_property_descriptor_fields (&prop_desc);
+}
```

Comment:
Ditto, init property descriptor, and remove comments, and use the returned value by `jerry_to_property_descriptor`.